File: src/forward_model/implementations/root.zig
const measurement = @import("../instrument_grid/grid_calculation/types.zig");
const instrument_implementation = @import("instrument/implementation.zig");

pub const Bindings = measurement.Implementations;

pub fn exact() Bindings {
    return .{
        .instrument = instrument_implementation.resolve("builtin.generic_response").?,
    };
}

File: rust_src/forward_model/implementations/root.rs
use crate::forward_model::{
    implementations::{instrument, noise, surface, transport},
    instrument_grid::grid_calculation::types,
};

pub type Bindings = types::Implementations;

pub fn exact() -> Bindings {
    Bindings {
        transport: transport::resolve(transport::DISPATCHER_ID)
            .expect("built-in transport provider must exist"),
        surface: surface::resolve(surface::LAMBERTIAN_SURFACE_ID)
            .expect("built-in surface provider must exist"),
        instrument: instrument::resolve(instrument::GENERIC_RESPONSE_ID)
            .expect("built-in instrument provider must exist"),
        noise: noise::resolve(noise::SCENE_NOISE_ID).expect("built-in noise provider must exist"),
    }
}
